src/test/java/org/mifos/integrationtest/cucumber/stepdef/EmailStepDef.java

```diff
@@ -0,0 +1,149 @@
+    public void theEmailShouldBeSentToWithSubjectAndBody(String subject, String body) throws Exception {
+        await().atMost(awaitMost, SECONDS).pollInterval(pollInterval, SECONDS).untilAsserted(() -> {
+            logger.info(String.valueOf(greenMail.getReceivedMessages().length));
+            assertThat(greenMail.getReceivedMessages().length == 1).isTrue();
+
+            String receivedTo = GreenMailUtil.getAddressList(greenMail.getReceivedMessages()[0].getAllRecipients());
+            String receivedSubject = greenMail.getReceivedMessages()[0].getSubject();
+            String receivedBody = GreenMailUtil.getBody(greenMail.getReceivedMessages()[0]);
+
+            assertThat(receivedTo).isNotNull();
+            assertThat(subject.equals(receivedSubject)).isTrue();
+            assertThat(body.equals(receivedBody)).isTrue();
+        });
+    }
+
+    @Then("I should be able to extract error from response")
+    public void iShouldBeAbleToExtractErrorFromResponse() {
+        assertThat(scenarioScopeState.response).isNotNull();
+        assertThat(scenarioScopeState.response).containsMatch("Bad Request");
+    }
+
+    @And("I can verify callback received with success")
+    public void iCanVerifyCallbackReceivedWithSuccess() {
+        await().atMost(awaitMost, SECONDS).pollInterval(pollInterval, SECONDS).untilAsserted(() -> {
+            boolean flag = false;
+            List<ServeEvent> allServeEvents = getAllServeEvents();
+            for (int i = allServeEvents.size() - 1; i >= 0; i--) {
+                ServeEvent request = allServeEvents.get(i);
+                if (!(request.getRequest().getBodyAsString()).isEmpty() && request.getRequest().getUrl().equals("/sendMail")) {
+                    String rootNode = null;
+                    rootNode = request.getRequest().getBodyAsString();
+                    logger.info("Rootnode value:" + rootNode);
+                    assertThat(rootNode.contains("Email sent successfully")).isTrue();
+                }
+            }
+
+        });
+    }
+
+    @And("I can verify callback received with failure")
+    public void iCanVerifyCallbackReceivedWithFailure() {
+        await().atMost(awaitMost, SECONDS).pollInterval(pollInterval, SECONDS).untilAsserted(() -> {
+            boolean flag = false;
+            List<ServeEvent> allServeEvents = getAllServeEvents();
+            for (int i = allServeEvents.size() - 1; i >= 0; i--) {
+                ServeEvent request = allServeEvents.get(i);
+                if (!(request.getRequest().getBodyAsString()).isEmpty() && request.getRequest().getUrl().equals("/sendMail")) {
+                    String rootNode = null;
+                    rootNode = request.getRequest().getBodyAsString();
+                    logger.info("Rootnode value:" + rootNode);
+                    assertThat(rootNode
+                            .contains("Email could not be sent to [recipient1@example.com] because of Mail server connection failed"))
```

Comment:
already mentioned in other PR, The error request body is not proper.
the message should be in sentence case